src/UI/Components/Diagram/init.ts

```diff
@@ -193,48 +194,56 @@ export default function diagramInit(
       target.id as dia.Cell.ID,
     ) as ServiceEntityBlock;
 
-    if (
-      sourceCell.get("isEmbedded") &&
-      sourceCell.get("isEmbeddedTo") !== null
-    ) {
-      sourceCell.set("isEmbeddedTo", targetCell.id);
-      didSourceChanged = true;
-    }
-    if (
-      targetCell.get("isEmbedded") &&
-      targetCell.get("isEmbeddedTo") !== null
-    ) {
-      targetCell.set("isEmbeddedTo", sourceCell.id);
-      didTargetChanged = true;
-    }
-
     const sourceRelations = sourceCell.getRelations();
     const targetRelations = targetCell.getRelations();
     const targetName = targetCell.getName();
     const sourceName = sourceCell.getName();
 
     if (sourceRelations) {
-      const doesSourceHaveRule = connectionRules[sourceName].find(
+      const sourceConnectionRule = connectionRules[sourceName].find(
         (rule) => rule.name === targetName,
       );
 
-      if (doesSourceHaveRule) {
-        sourceCell.addRelation(targetCell.id as string, targetName);
+      if (sourceConnectionRule) {
+        sourceCell.addRelation(
+          targetCell.id as string,
+          sourceConnectionRule.attributeName as string,
+        );
         didSourceChanged = true;
+        didConnectionWasSet = true;
       }
     }
 
     if (targetRelations) {
-      const doesTargetHaveRule = connectionRules[targetName].find(
+      const targetConnectionRule = connectionRules[targetName].find(
         (rule) => rule.name === sourceName,
       );
-
-      if (doesTargetHaveRule) {
-        targetCell.addRelation(sourceCell.id as string, sourceName);
+      if (targetConnectionRule) {
+        targetCell.addRelation(
+          sourceCell.id as string,
+          targetConnectionRule.attributeName as string,
+        );
         didTargetChanged = true;
+        didConnectionWasSet = true;
       }
     }
 
+    if (!didConnectionWasSet) {
```

Comment:
You have a lot of repetition in here, which makes it hard to read. Why not extract the logic into a function? I quickly drafted it, you might need to make adjustments and add typings
```
const checkUpdated = (element) => {
    const cell = graph.getCell(
        element.id as dia.Cell.ID,
    ) as ServiceEntityBlock;

    const relations = cell.getRelations();
    const name = cell.getName();

    if (relations) {
        const elementConnectionRule = connectionRules[name].find((rule) => rule.name === name);
        if (elementConnectionRule) {
            cell.addRelation(
                sourceCell.id as string,
                targetConnectionRule.attributeName as string,
              );
            updateInstancesToSend(cell, ActionEnum.UPDATE);
            return; // you don't need further information, so you can early return the function
        }
    }

    if (cell.get("isEmbedded") &&
    cell.get("embeddedTo") !== null) {
        cell.set("embeddedTo", cell.id);
        updateInstancesToSend(cell, ActionEnum.UPDATE);
        return;
    }
}

const source = linkView.model.source();
const target = linkView.model.target();
checkUpdated(source);
checkUpdated(target);

```